python/ray/data/read_api.py

```diff
@@ -2285,6 +2286,79 @@ def get_dbutils():
     )
 
 
+@PublicAPI
+def read_iceberg(
+    *,
+    table_identifier: str,
+    row_filter: Union[str, "BooleanExpression"] = None,
+    concurrency: int = -1,
+    selected_fields: Tuple[str, ...] = ("*",),
+    snapshot_id: Optional[int] = None,
+    scan_kwargs: Optional[Dict[str, str]] = None,
+    catalog_kwargs: Optional[Dict[str, str]] = None,
+    ray_remote_args: Optional[Dict[str, Any]] = None,
+) -> Dataset:
+    """Create a :class:`~ray.data.Dataset` from an Iceberg table. The table to read
+    from is specified using a fully qualified ``table_identifier``. Using PyIceberg,
+    any intended row filters, snapshot IDs, etc. are applied, and the files that
+    satisfy the query are distributed across Ray read tasks. The number of tasks is
+    determined by ``parallelism`` which can be requested from this interface or
+    automatically chosen if unspecified (see the``parallelism`` arg below).
```

Comment:
nit to fix the doc format
```suggestion
    automatically chosen if unspecified (see the ``parallelism`` arg below).
```